src/CollisionHandler.cpp

```diff
@@ -0,0 +1,45 @@
+#include "CollisionHandler.h"
+#include "WorldMap.h"
+#include <math.h>
+
+sf::Vector2f CollisionHandler::handleTileCollision(const sf::FloatRect & entityAABB, const WorldMap & worldMap)
+{
+	sf::Vector2f newPos;
+	for (const auto& tileAABB : worldMap.getCollidableTileLayer().getTileMap())
+	{
+		sf::FloatRect intersection;
+		if (!entityAABB.intersects(tileAABB, intersection))
+		{
+			continue;
+		}
+
+		//Determine which axis collisions happens
+		const float xDifference = (entityAABB.left + (entityAABB.width / 2.0f)) - (tileAABB.left + (tileAABB.width / 2.0f));
+		const float yDifference = (entityAABB.top + (entityAABB.height / 2.0f)) - (tileAABB.top + (tileAABB.height / 2.0f));
```

Comment:
We actually want to move the player in the opposite direction of their movement.  See code segments:  http://codepad.org/dSzlMhHG and http://codepad.org/BrjrMenZ.  Also assumes non-diagonal movement